magma/bits.py

```diff
@@ -627,6 +646,16 @@ def __lt__(self, other):
 class SInt(Int):
     hwtypes_T = ht.SIntVector
 
+    @debug_wire
+    def wire(self, other, debug_info=None):
+        if isinstance(other, UInt):
```

Comment:
it seems redundant to have logic `is_wireable` and then do this check again in `wire()`. Seems like `wire()` for any type should by default call `is_wireable()` and the error message of `"Cannot wire..."` is generic.

TL;DR: should we just call `is_wireable()` by default before any logic in `wire()`?

Reply:
Ok, moved the wire error checking logic to wire() and updated to use is_wireable.  Overall I think it's an improvement, it consolidates the logic into a clear API, whereas before the error checking was done in various parts of the wiring pipeline.  As a side effect, it seems some error messages are improved, whereas other are less precise (since we early error before wiring things by checking is_wireable, we don't descend into the wiring pipeline where old errors were being raise).  I'll work on moving the old error checking and message logic up into the is_wireable flow (I think we can augment it to return an object that indicates and error with a message, this way we can report the generic error "Cannot wire X to Y" and then add a more specific error below it (e.g. array sizes do not match).